File: app/src/main/java/com/p1/uber/provides/ClientBookingProvider.java
package com.p1.uber.provides;

import com.google.android.gms.tasks.Task;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.p1.uber.modelos.ClientBooking;

import java.util.HashMap;
import java.util.Map;

public class ClientBookingProvider {

    private DatabaseReference mDatabase;

    public ClientBookingProvider (){
        mDatabase= FirebaseDatabase.getInstance().getReference().child("ClientBooking");

    }

    public Task<Void> create(ClientBooking clientBooking){
        return mDatabase.child(clientBooking.getIdClient()).setValue(clientBooking);
    }

    public Task<Void> updateStatus(String IdClientBooking, String status){

        Map<String , Object> map = new HashMap<>();
        map.put("status", status);
        return mDatabase.child(IdClientBooking).updateChildren(map);

    }

    public Task<Void> updateIdHistoryBooking(String IdClientBooking){
        String idPush = mDatabase.push().getKey();
        Map<String , Object> map = new HashMap<>();
        map.put("idHistoryBooking", idPush);
        return mDatabase.child(IdClientBooking).updateChildren(map);

    }

    public  DatabaseReference getstatus(String idClientBooking) {
        return mDatabase.child(idClientBooking).child("status");
    }

    public  DatabaseReference getClientBooking(String idClientBooking) {
        return mDatabase.child(idClientBooking);
    }

    public  Task<Void> delete(String IdClientBooking){
        return mDatabase.child(IdClientBooking).removeValue();
    }
}

File: app/src/main/java/com/p1/uberfares/provides/ClientBookingProvider.kt
package com.p1.uberfares.provides

import android.os.Build
import androidx.annotation.RequiresApi
import com.google.android.gms.tasks.Task
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.p1.uberfares.modelos.ClientBooking
import java.time.LocalTime

class ClientBookingProvider {
    private val mDatabase: DatabaseReference

    init {
        mDatabase = FirebaseDatabase.getInstance().reference.child("ClientBooking")
    }

    fun create(clientBooking: ClientBooking): Task<Void> {
        return mDatabase.child(clientBooking.idClient!!).setValue(clientBooking)
    }

    @RequiresApi(Build.VERSION_CODES.O)
    fun updateStatus(IdClientBooking: String?, status: String): Task<Void> {
        val map: MutableMap<String, Any> = HashMap()
        map["status"] = status

        return mDatabase.child(IdClientBooking!!).updateChildren(map)
    }

    @RequiresApi(Build.VERSION_CODES.O)
    fun updateStatus2(IdClientBooking: String?, endTime: String): Task<Void> {
        val map: MutableMap<String, Any> = HashMap()
        map["idDriver"] = endTime
        return mDatabase.child(IdClientBooking!!).updateChildren(map)
    }

    @RequiresApi(Build.VERSION_CODES.O)
    fun updateStatus3(IdClientBooking: String?, starTtime: LocalTime): Task<Void> {
        val map: MutableMap<String, Any> = HashMap()
        map["starttime"] = starTtime
        return mDatabase.child(IdClientBooking!!).updateChildren(map)
    }

    fun updateIdHistoryBooking(IdClientBooking: String?): Task<Void> {
        val idPush = mDatabase.push().key
        val map: MutableMap<String, Any?> = HashMap()
        map["idHistoryBooking"] = idPush
        return mDatabase.child(IdClientBooking!!).updateChildren(map)
    }

    fun getstatus(idClientBooking: String?): DatabaseReference {
        return mDatabase.child(idClientBooking!!).child("status")
    }

    fun getClientBooking(idClientBooking: String?): DatabaseReference {
        return mDatabase.child(idClientBooking!!)
    }

    fun delete(IdClientBooking: String?): Task<Void> {
        return mDatabase.child(IdClientBooking!!).removeValue()
    }
}